contracts/MemeFactory.sol

```diff
@@ -582,16 +536,8 @@ abstract contract MemeFactory {
         require(_locked == 1, "Reentrancy guard");
         _locked = 2;
 
-        uint256 amount = scheduledForAscendance;
+        uint256 amount = _launched > 0 ? scheduledForAscendance : _launchCampaign();
```

Comment:
This is a major update and is much cleaner. Either return `scheduledForAscendance` or launch campaign based on `scheduledForAscendance` as well.